### PR TITLE
Updated descriptor to use `RpcType` instead of `KType` directly

### PR DESCRIPTION
**Pull request by Mr3zee**
**Subsystem**
Core

**Problem Description**
KType maybe changes in the future to KClass or custom metadata

**Solution**
Introduce intermediate `RpcType`

### CODE REVIEW

File: compiler-plugin/compiler-plugin-backend/src/main/core/kotlinx/rpc/codegen/extension/RpcIrContext.kt [hunk 1]
```diff
@@ -105,6 +105,10 @@ internal class RpcIrContext(
         getRpcIrClassSymbol("RpcServiceDescriptor", "descriptor")
     }
 
+    val rpcType by lazy {
```
Comment (e5l) [inline compiler-plugin/compiler-plugin-backend/src/main/core/kotlinx/rpc/codegen/extension/RpcIrContext.kt:108]: Should we extract type constants to the enum?

File: gradle-conventions/latest-only/src/main/kotlin/conventions-gradle-doctor.gradle.kts [hunk 2]
```diff
@@ -9,6 +9,6 @@ plugins {
 doctor {
     enableTestCaching = false
     warnWhenNotUsingParallelGC = true
-    disallowMultipleDaemons = true
+    disallowMultipleDaemons = false
```
Comment (e5l) [inline gradle-conventions/latest-only/src/main/kotlin/conventions-gradle-doctor.gradle.kts:12]: memory💦 ?

**Review by e5l — APPROVED** [review]

**Comment by e5l** [comment]
Nice, please check CI before merging

File: compiler-plugin/compiler-plugin-backend/src/main/core/kotlinx/rpc/codegen/extension/RpcIrContext.kt (hunk 1 above)
Comment (Mr3zee) [reply compiler-plugin/compiler-plugin-backend/src/main/core/kotlinx/rpc/codegen/extension/RpcIrContext.kt:108]: Maybe, but definitely later. It works now, and I don't have problems maintaining it.

**Pull request merged**